src/vs/workbench/parts/debug/browser/debugStatus.ts

```diff
@@ -57,6 +66,29 @@ export class DebugStatus extends Themable implements IStatusbarItem {
 				}
 			}
 		}));
+
+		this.toDispose.push(viewletService.onDidViewletOpen(e => {
+			this.viewletId = e.getId();
+			if (this.showInStatusBar === 'whenViewHidden') {
+				//Show status bar item if new view is not debug
+				if (this.viewletId !== VIEWLET_ID) {
+					if (!this.statusBarItem) {
+						this.doRender();
```

Comment:
No need for this case imho. The onDidViewletClose will catch it.
You should just check if openeing a viewlet shuold hide the statusbar